### ISSUE TITLE
Compilation issue with debian unstable (trixie)

### ISSUE THREAD

**Issue by colisee** [Opendigitalradio/ODR-AudioEnc#26]
Environment: [schroot with unstable](https://wiki.debian.org/Packaging/Pre-Requisites#Option_2:_Schroot_and_Sbuild)

Output from make
```
  CXX      contrib/odr_audioenc-ReedSolomon.o
contrib/ReedSolomon.cpp: In member function 'int ReedSolomon::encode(void*, void*, size_t)':
contrib/ReedSolomon.cpp:81:5: error: 'uint8_t' was not declared in this scope
   81 |     uint8_t* input = reinterpret_cast<uint8_t*>(data);
      |     ^~~~~~~
contrib/ReedSolomon.cpp:39:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   38 | #include <assert.h>
  +++ |+#include <cstdint>
   39 |
```

Remarks:
- I followed the compilation instructions in README.md
- The compilation works for schroot based on bullseye and bookworm
- I have the same issue when building the debian package for unstable
- It seems that debian unstable introduced cpp-13 (this is absent from debian bookworm)

**Comment by colisee** [comment]
https://gitlab.onelab.info/gmsh/gmsh/-/issues/2416

Some C++ Standard Library headers have been changed to no longer include other headers that were being used internally by the library.

**Comment by colisee** [comment]
Hi @mpbraendli,

I think I have found the explanation:

1. include dependency path
   contrib/ReedSolomon.cpp - stdexcept - string - bits/char_traits.h - cstdint

1. Header file <bits/char_traits.h>
   bullseye:
   ```
   #if __cplusplus >= 201103L
     #include <cstdint>
   ```

   unstable:
   ```
   #if __cplusplus >= 201103L
     # include <type_traits>
     #if !defined __UINT_LEAST16_TYPE__ || !defined __UINT_LEAST32_TYPE__
       # include <cstdint>
     #endif
   #endif
   ```

Somehow, __UINT_LEAST16_TYPE__ and/or __UINT_LEAST32_TYPE__ are defined in unstable, preventing cstdint from being included by the preprocessor.

Anyhow, I believe that `<cstdint>` should be added to source file `contrib/ReedSolomon.cpp` since the latter explicitely refers to type `uint8_t`

In order to push the package to debian, I would need this change to be included in a new release...

**Comment by mpbraendli** [comment]
Hi, yes the header `#include` is missing. Same issue popped up in ODR-DabMux too.

I need to do releases for all tools that have ReedSolomon.cpp

Thanks for the reminder

**Comment by colisee** [comment]
Merci @mpbraendli 

By the way, I confirm that once you add `#include <cstdint>` to ReedSolomon.cpp, the compilation is still OK in bullseye (and probably bookworm, although I didn't check, yet)

**Comment by mpbraendli** [comment]
Fixed in v3.4.0
